xdsl/dialects/stablehlo.py

```diff
@@ -18,18 +19,41 @@
     result_def,
 )
 
+# region Abstract Base Classes
+
+
+class ElementwiseBinaryOperation(IRDLOperation, abc.ABC):
+    # TODO: Remove this constraint for complex types.
+    T = Annotated[AnyTensorType, ConstraintVar("T")]
+
```

Comment:
Should these have some verification that the inputs are "numeric"

Reply:
Eventually we'll make these constrained to the values that the original stablehlo dialect supports, I don't think it's critical right now.